registry/remote/credentials/internal/config/config.go

```diff
@@ -160,13 +160,18 @@ func (cfg *Config) GetCredential(serverAddress string) (auth.Credential, error)
 	cfg.rwLock.RLock()
 	defer cfg.rwLock.RUnlock()
 
-	authCfgBytes, ok := cfg.authsCache[serverAddress]
+	return getCredentialFromCache(cfg.authsCache, serverAddress)
+}
+
+// getCredentialsFromCache is a helper function to get the credential for serverAddress from authsCache in a raw format.
+func getCredentialFromCache(authsCache map[string]json.RawMessage, serverAddress string) (auth.Credential, error) {
```

Comment:
We can refactor this in another way using generics, so that it can be reused by both config and readonly config:
Something like:
```go
func matchAuth[V any](auths map[string]V, serverAddress string) (V, bool) {
	if v, ok := auths[serverAddress]; ok {
		return v, true
	}
	for addr, v := range auths {
		if toHostname(addr) == serverAddress {
			return v, true
		}
	}
	var zero V
	return zero, false
}
```

Reply:
It's currently used by both stores. For readonly in `LoadFromReader` where I perform parsing all entries to `auth.Credential` because they won't change during the lifetime of such a config, it can be done here. There is info whether the whole config is valid and no computation is needed for getting a credential.

For file backedd config it's used in method `GetCredential`. They need to be parsed and looked up on each call because they may change.